contributors/design-proposals/auth/bound-service-account-tokens.md

```diff
@@ -213,6 +213,67 @@ The token payload will be:
 }
 ```
 
+### Add Service Account signing key retrieval API
+
+A non-resource API will be added to the apiserver that will expose an [OIDC
+discovery](https://openid.net/specs/openid-connect-discovery-1_0.html)
+compatible configuration. This will allow clients to retrieve a list of
+currently valid signing keys as an array of JWKs. For example, if the service
+account issuer is configured with the name `https://dev.cluster.internal`, the
+apiserver could expose the following configuration:
+
+```
+> GET /.well-known/openid-configuration
+{
+  "issuer": "https://dev.cluster.internal",
+  "jwks_uri": "https://dev.cluster.internal/serviceaccountkeys/v1",
+  "authorization_endpoint": "urn:kubernetes:programmatic_authorization",
+  "response_types_supported": [
+    "id_token"
+  ],
+  "subject_types_supported": [
+    "public"
+  ],
+  "id_token_signing_alg_values_supported": [
+    "RS256",
+    "ES256"
+  ],
+  "claims_supported": [
+    "sub",
+    "iss"
+  ]
```

Comment:
if `grant_types` is omitted, the default value is `["authorization_code", "implicit"]`, which means that `token_endpoint` is also required

Reply:
using an openid discovery doc as a vehicle for publishing public keys gets weird when the tokens we're wanting to allow someone to validate can't actually be obtained via an openid flow... we'd have to do things like advertise `grant_types_supported:[]` and make our `authorization_endpoint` point to a URL that rejected all requests. that seems pretty weird.

Reply:
However, grant types are extensible:

https://tools.ietf.org/html/rfc6749#section-8.3

An alternative is to make `"grant_types":["urn:kubernetes:grant_type:programmatic_authorization"]` and make `authorization_endpoint` point somewhere bogus.

Reply:
@BrenoDeMedeiros do you have any thoughts on how we should express the Kubernetes id_token flow in a openid-configuration to support oidc discovery?